packages/react-native/React/Fabric/Mounting/ComponentViews/ScrollView/RCTScrollViewComponentView.mm

```diff
@@ -149,6 +152,104 @@ - (void)dealloc
   [self.scrollViewDelegateSplitter removeAllDelegates];
 }
 
+- (void)_registerKeyboardListener
+{
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(_keyboardWillChangeFrame:)
+                                               name:UIKeyboardWillChangeFrameNotification
+                                             object:nil];
+}
+
+- (void)_unregisterKeyboardListener
+{
+  [[NSNotificationCenter defaultCenter] removeObserver:self name:UIKeyboardWillChangeFrameNotification object:nil];
+}
```

Comment:
this seems like it is never called. This might make the RCTScrollView leak.

We should probably call it in the `deinit` but maybe also in the `prepareForRecycle`. In that last case, we need to figure out when to register the listener again.